ir/TypeAndName.cpp

```diff
@@ -0,0 +1,12 @@
+#include "TypeAndName.h"
+
+TypeAndName::TypeAndName(std::string name, std::string type)
+        : name(std::move(name)), type(std::move(type)) {}
+
+std::string TypeAndName::getType() const {
+    return type;
```

Comment:
Types are stored in multiple classes, so it would be difficult to make sure that there is `handleReservedWords` everywhere. I added `handleReservedWords` calls to `TreeVisitor`, so IR will always contain correct type names.
Issue for tests: #22

Reply:
Good point, forgot that this method is used in other cases than just emitting code.